### PR TITLE
K8SPXC-1368: PVC: forward dataSource field

### PR DESCRIPTION
**Pull request by ebuildy**
[K8SPXC-1368](https://jira.percona.com/browse/K8SPXC-1368) [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=percona&utm_campaign=badge_info)

**CHANGE DESCRIPTION**
---
**Problem:**
*Short explanation of the problem.*

I want to clone PVC from a VolumeSnapshot (containing a xtrabackup).

**Cause:**
*Short explanation of the root cause of the issue if applicable.*

Can be useful to be able to clone PVC from other PVC or VolumeSnapshot.

**Solution:**
*Short explanation of the solution we are providing with this PR.*

CRD define dataSource, dataSourceRef fields, this PR implement it. 

As a workaround, I must create the PVC manually before deploying the pxc-cluster, which is not very handy.

**CHECKLIST**
---
**Jira**
- [ ] Is the Jira ticket created and referenced properly?
- [ ] Does the Jira ticket have the proper statuses for documentation (`Needs Doc`) and QA (`Needs QA`)?
- [ ] Does the Jira ticket link to the proper milestone (Fix Version field)?

**Tests**
- [ ] Is an E2E test/test case added for the new feature/change?
- [ ] Are unit tests added where appropriate?
- [ ] Are OpenShift compare files changed for E2E tests (`compare/*-oc.yml`)?

**Config/Logging/Testability**
- [ ] Are all needed new/changed options added to default YAML files?
- [ ] Did we add proper logging messages for operator actions?
- [ ] Did we ensure compatibility with the previous version or cluster upgrade process?
- [ ] Does the change support oldest and newest supported PXC version?
- [ ] Does the change support oldest and newest supported Kubernetes version?

[K8SPXC-1368]: https://perconadev.atlassian.net/browse/K8SPXC-1368?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| affinity-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1624/41cb00b0/affinity-8-0.log) |
| auto-tuning-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1624/41cb00b0/auto-tuning-8-0.log) |
| cross-site-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1624/41cb00b0/cross-site-8-0.log) |
| demand-backup-cloud-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1624/41cb00b0/demand-backup-cloud-8-0.log) |
| demand-backup-encrypted-with-tls-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1624/41cb00b0/demand-backup-encrypted-with-tls-8-0.log) |
| demand-backup-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1624/41cb00b0/demand-backup-8-0.log) |
| haproxy-5-7 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1624/41cb00b0/haproxy-5-7.log) |
| haproxy-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1624/41cb00b0/haproxy-8-0.log) |
| init-deploy-5-7 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1624/41cb00b0/init-deploy-5-7.log) |
| init-deploy-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1624/41cb00b0/init-deploy-8-0.log) |
| limits-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1624/41cb00b0/limits-8-0.log) |
| monitoring-2-0-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1624/41cb00b0/monitoring-2-0-8-0.log) |
| one-pod-5-7 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1624/41cb00b0/one-pod-5-7.log) |
| one-pod-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1624/41cb00b0/one-pod-8-0.log) |
| pitr-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1624/41cb00b0/pitr-8-0.log) |
| pitr-gap-errors-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1624/41cb00b0/pitr-gap-errors-8-0.log) |
| proxy-protocol-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1624/41cb00b0/proxy-protocol-8-0.log) |
| proxysql-sidecar-res-limits-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1624/41cb00b0/proxysql-sidecar-res-limits-8-0.log) |
| pvc-resize-5-7 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1624/41cb00b0/pvc-resize-5-7.log) |
| pvc-resize-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1624/41cb00b0/pvc-resize-8-0.log) |
| recreate-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1624/41cb00b0/recreate-8-0.log) |
| restore-to-encrypted-cluster-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1624/41cb00b0/restore-to-encrypted-cluster-8-0.log) |
| scaling-proxysql-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1624/41cb00b0/scaling-proxysql-8-0.log) |
| scaling-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1624/41cb00b0/scaling-8-0.log) |
| scheduled-backup-5-7 | [failure](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1624/41cb00b0/scheduled-backup-5-7.log) |
| scheduled-backup-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1624/41cb00b0/scheduled-backup-8-0.log) |
| security-context-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1624/41cb00b0/security-context-8-0.log) |
| smart-update1-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1624/41cb00b0/smart-update1-8-0.log) |
| smart-update2-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1624/41cb00b0/smart-update2-8-0.log) |
| storage-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1624/41cb00b0/storage-8-0.log) |
| tls-issue-cert-manager-ref-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1624/41cb00b0/tls-issue-cert-manager-ref-8-0.log) |
| tls-issue-cert-manager-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1624/41cb00b0/tls-issue-cert-manager-8-0.log) |
| tls-issue-self-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1624/41cb00b0/tls-issue-self-8-0.log) |
| upgrade-consistency-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1624/41cb00b0/upgrade-consistency-8-0.log) |
| upgrade-haproxy-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1624/41cb00b0/upgrade-haproxy-8-0.log) |
| upgrade-proxysql-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1624/41cb00b0/upgrade-proxysql-8-0.log) |
| users-5-7 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1624/41cb00b0/users-5-7.log) |
| users-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1624/41cb00b0/users-8-0.log) |
| validation-hook-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1624/41cb00b0/validation-hook-8-0.log) |
| We run 39 out of 39|

commit: https://github.com/percona/percona-xtradb-cluster-operator/pull/1624/commits/41cb00b08b70b681eaadf85d5576188bbe25cfaf
image: `perconalab/percona-xtradb-cluster-operator:PR-1624-41cb00b0`

**Comment by egegunes** [comment]
@ebuildy thank you for the contribution. I need to test this.

**Comment by egegunes** [comment]
@inelpandzic if you have time please test this PR, I couldn't find the time to do it.

in most basic way you can do the following: create a cluster `cluster1`, write some data, delete cluster without deleting secrets or PVCs and create cluster `cluster2` using `datadir-cluster1-pxc-0` as `DataSource` (don't forget to use the same credentials from `cluster1`) and check if the data is in `cluster2`.

**Comment by inelpandzic** [comment]
@ebuildy @egegunes I have tested this and works as expected. I think this is good.

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| affinity-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1624/d283c39d/affinity-8-0.log) |
| auto-tuning-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1624/d283c39d/auto-tuning-8-0.log) |
| cross-site-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1624/d283c39d/cross-site-8-0.log) |
| demand-backup-cloud-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1624/d283c39d/demand-backup-cloud-8-0.log) |
| demand-backup-encrypted-with-tls-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1624/d283c39d/demand-backup-encrypted-with-tls-8-0.log) |
| demand-backup-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1624/d283c39d/demand-backup-8-0.log) |
| haproxy-5-7 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1624/d283c39d/haproxy-5-7.log) |
| haproxy-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1624/d283c39d/haproxy-8-0.log) |
| init-deploy-5-7 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1624/d283c39d/init-deploy-5-7.log) |
| init-deploy-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1624/d283c39d/init-deploy-8-0.log) |
| limits-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1624/d283c39d/limits-8-0.log) |
| monitoring-2-0-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1624/d283c39d/monitoring-2-0-8-0.log) |
| one-pod-5-7 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1624/d283c39d/one-pod-5-7.log) |
| one-pod-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1624/d283c39d/one-pod-8-0.log) |
| pitr-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1624/d283c39d/pitr-8-0.log) |
| pitr-gap-errors-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1624/d283c39d/pitr-gap-errors-8-0.log) |
| proxy-protocol-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1624/d283c39d/proxy-protocol-8-0.log) |
| proxysql-sidecar-res-limits-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1624/d283c39d/proxysql-sidecar-res-limits-8-0.log) |
| pvc-resize-5-7 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1624/d283c39d/pvc-resize-5-7.log) |
| pvc-resize-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1624/d283c39d/pvc-resize-8-0.log) |
| recreate-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1624/d283c39d/recreate-8-0.log) |
| restore-to-encrypted-cluster-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1624/d283c39d/restore-to-encrypted-cluster-8-0.log) |
| scaling-proxysql-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1624/d283c39d/scaling-proxysql-8-0.log) |
| scaling-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1624/d283c39d/scaling-8-0.log) |
| scheduled-backup-5-7 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1624/d283c39d/scheduled-backup-5-7.log) |
| scheduled-backup-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1624/d283c39d/scheduled-backup-8-0.log) |
| security-context-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1624/d283c39d/security-context-8-0.log) |
| smart-update1-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1624/d283c39d/smart-update1-8-0.log) |
| smart-update2-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1624/d283c39d/smart-update2-8-0.log) |
| storage-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1624/d283c39d/storage-8-0.log) |
| tls-issue-cert-manager-ref-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1624/d283c39d/tls-issue-cert-manager-ref-8-0.log) |
| tls-issue-cert-manager-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1624/d283c39d/tls-issue-cert-manager-8-0.log) |
| tls-issue-self-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1624/d283c39d/tls-issue-self-8-0.log) |
| upgrade-consistency-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1624/d283c39d/upgrade-consistency-8-0.log) |
| upgrade-haproxy-5-7 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1624/d283c39d/upgrade-haproxy-5-7.log) |
| upgrade-haproxy-8-0 | [failure](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1624/d283c39d/upgrade-haproxy-8-0.log) |
| upgrade-proxysql-5-7 | [failure](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1624/d283c39d/upgrade-proxysql-5-7.log) |
| upgrade-proxysql-8-0 | [failure](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1624/d283c39d/upgrade-proxysql-8-0.log) |
| users-5-7 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1624/d283c39d/users-5-7.log) |
| users-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1624/d283c39d/users-8-0.log) |
| validation-hook-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1624/d283c39d/validation-hook-8-0.log) |
| We run 41 out of 41|

commit: https://github.com/percona/percona-xtradb-cluster-operator/pull/1624/commits/d283c39d6b7e2521053e18551101cce119cf0d08
image: `perconalab/percona-xtradb-cluster-operator:PR-1624-d283c39d`

**Review by inelpandzic — APPROVED** [review]

File: pkg/pxc/app/pvc.go [hunk 1]
```diff
@@ -24,5 +24,7 @@ func VolumeSpec(vspec *api.VolumeSpec) corev1.PersistentVolumeClaimSpec {
 		StorageClassName: vspec.PersistentVolumeClaim.StorageClassName,
 		AccessModes:      vspec.PersistentVolumeClaim.AccessModes,
 		Resources:        vspec.PersistentVolumeClaim.Resources,
+		DataSource:       vspec.PersistentVolumeClaim.DataSource,
```
Comment (hors) [inline pkg/pxc/app/pvc.go:27]: @ebuildy could you please add these fields to our template cr.yaml like it was done for e.g. storageClassName https://github.com/percona/percona-xtradb-cluster-operator/blob/main/deploy/cr.yaml#L179 and we will accept your PR. Thanks!

Comment (ebuildy) [reply pkg/pxc/app/pvc.go:27]: Dobryj den’ @hors , hope you are doing well, thanks you for your review, I have just updated the cr.yaml file.

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| affinity-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1624/d6e85bca/affinity-8-0.log) |
| auto-tuning-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1624/d6e85bca/auto-tuning-8-0.log) |
| cross-site-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1624/d6e85bca/cross-site-8-0.log) |
| demand-backup-cloud-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1624/d6e85bca/demand-backup-cloud-8-0.log) |
| demand-backup-encrypted-with-tls-8-0 | [failure](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1624/d6e85bca/demand-backup-encrypted-with-tls-8-0.log) |
| demand-backup-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1624/d6e85bca/demand-backup-8-0.log) |
| haproxy-5-7 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1624/d6e85bca/haproxy-5-7.log) |
| haproxy-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1624/d6e85bca/haproxy-8-0.log) |
| init-deploy-5-7 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1624/d6e85bca/init-deploy-5-7.log) |
| init-deploy-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1624/d6e85bca/init-deploy-8-0.log) |
| limits-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1624/d6e85bca/limits-8-0.log) |
| monitoring-2-0-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1624/d6e85bca/monitoring-2-0-8-0.log) |
| one-pod-5-7 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1624/d6e85bca/one-pod-5-7.log) |
| one-pod-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1624/d6e85bca/one-pod-8-0.log) |
| pitr-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1624/d6e85bca/pitr-8-0.log) |
| pitr-gap-errors-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1624/d6e85bca/pitr-gap-errors-8-0.log) |
| proxy-protocol-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1624/d6e85bca/proxy-protocol-8-0.log) |
| proxysql-sidecar-res-limits-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1624/d6e85bca/proxysql-sidecar-res-limits-8-0.log) |
| pvc-resize-5-7 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1624/d6e85bca/pvc-resize-5-7.log) |
| pvc-resize-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1624/d6e85bca/pvc-resize-8-0.log) |
| recreate-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1624/d6e85bca/recreate-8-0.log) |
| restore-to-encrypted-cluster-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1624/d6e85bca/restore-to-encrypted-cluster-8-0.log) |
| scaling-proxysql-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1624/d6e85bca/scaling-proxysql-8-0.log) |
| scaling-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1624/d6e85bca/scaling-8-0.log) |
| scheduled-backup-5-7 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1624/d6e85bca/scheduled-backup-5-7.log) |
| scheduled-backup-8-0 | [failure](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1624/d6e85bca/scheduled-backup-8-0.log) |
| security-context-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1624/d6e85bca/security-context-8-0.log) |
| smart-update1-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1624/d6e85bca/smart-update1-8-0.log) |
| smart-update2-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1624/d6e85bca/smart-update2-8-0.log) |
| storage-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1624/d6e85bca/storage-8-0.log) |
| tls-issue-cert-manager-ref-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1624/d6e85bca/tls-issue-cert-manager-ref-8-0.log) |
| tls-issue-cert-manager-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1624/d6e85bca/tls-issue-cert-manager-8-0.log) |
| tls-issue-self-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1624/d6e85bca/tls-issue-self-8-0.log) |
| upgrade-consistency-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1624/d6e85bca/upgrade-consistency-8-0.log) |
| upgrade-haproxy-5-7 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1624/d6e85bca/upgrade-haproxy-5-7.log) |
| upgrade-haproxy-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1624/d6e85bca/upgrade-haproxy-8-0.log) |
| upgrade-proxysql-5-7 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1624/d6e85bca/upgrade-proxysql-5-7.log) |
| upgrade-proxysql-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1624/d6e85bca/upgrade-proxysql-8-0.log) |
| users-5-7 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1624/d6e85bca/users-5-7.log) |
| users-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1624/d6e85bca/users-8-0.log) |
| validation-hook-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1624/d6e85bca/validation-hook-8-0.log) |
| We run 41 out of 41|

commit: https://github.com/percona/percona-xtradb-cluster-operator/pull/1624/commits/d6e85bcab4452517fd8d8fab2fe279969120c12d
image: `perconalab/percona-xtradb-cluster-operator:PR-1624-d6e85bca`

File: deploy/cr.yaml [hunk 2]
```diff
@@ -178,6 +178,10 @@ spec:
       persistentVolumeClaim:
 #        storageClassName: standard
 #        accessModes: [ "ReadWriteOnce" ]
+#        dataSource:
+#          name: new-snapshot-test
+#          kind: VolumeSnapshot
+#          apiGroup: snapshot.storage.k8s.io
```
Comment (egegunes) [inline deploy/cr.yaml:184]: @inelpandzic we didn't test this with a `VolumeSnapshot` right? I think we need to do it before merging this.

Comment (inelpandzic) [reply deploy/cr.yaml:184]: @egegunes as expected, `VolumeSnapshot` also work. We can merge this.

**Review by inelpandzic — APPROVED** [review]

**Review by egegunes — APPROVED** [review]

**Comment by egegunes** [comment]
I don't know why CLA check is pending, @hors please merge when you have the chance.

**Comment by ebuildy** [comment]
> I don't know why CLA check is pending, @hors please merge when you have the chance.

when clicking on CLA check:

Capture d’écran, le 2024-03-29 à 11 37 24

**Review by hors — APPROVED** [review]

**Pull request merged**